streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -919,20 +987,16 @@ void validate(final Set<String> copartitionGroup,
                 if (!allRepartitionTopicsNumPartitions.containsKey(topic)) {
                     final Integer partitions = metadata.partitionCountForTopic(topic);
 
-                    if (partitions == null) {
-                        throw new org.apache.kafka.streams.errors.TopologyException(String.format("%sTopic not found: %s", logPrefix, topic));
-                    }
-
+                    final String[] topics = copartitionGroup.toArray(new String[copartitionGroup.size()]);
+                    Arrays.sort(topics);
                     if (numPartitions == UNKNOWN) {
+                        if (partitions == null) {
+                            throw new org.apache.kafka.streams.errors.TopologyException(String.format("%sTopics not co-partitioned: [%s]", logPrefix, Utils.join(Arrays.asList(topics), ",")));
```

Comment:
Same as above, let's add a FATAL error and throw a runtime exception like IllegalStateException.